content/post/124-protoschool-libp2p.md

```diff
@@ -0,0 +1,39 @@
+---
+date: 2021-02-15
+url: /2021-02-15-libp2p-comes-to-protoschool/
+title: libp2p Comes to ProtoSchool
+author: Teri Chadbourne
+tags: protoschool, libp2p, ipfs
+header_image: 124-libp2p-comes-to-protoschool.png
+---
+
+The [ProtoSchool](https://proto.school) team is pleased to announce the launch of a shiny new multiple-choice tutorial [introducing libp2p](https://proto.school/introducing-libp2p), the modular peer-to-peer networking stack that spun out of IPFS into its own first-class project and a dependency of IPFS itself. Today, IPFS focuses on _content addressing_ (i.e. finding, fetching and authenticating any piece of content in the web) while libp2p focuses on _process addressing_ (i.e., finding, connecting and authenticating any data transfer processes in the network).
+
+<div class="flex justify-center items-start mt4 mb4">
+    <img class="w-75-ns w-100" src="/124-libp2p-comes-to-protoschool/libp2p-ipfs-animation.gif" alt="libp2p animation representing libp2p being extracted out of ipfs" />
+</div>
+
+Historically, companies that produced peer-to-peer applications like like Skype or BitTorrent created their own protocols to support them. Those protocols made a lot of assumptions about the environments they'd run in and the needs they'd meet, making them incredibly hard to upgrade or adapt to new environments. libp2p, however, has been extracted from its original implementation as the networking stack of IPFS, and can now serve a broad variety of use cases. While everyone building on IPFS or Filecoin is using libp2p as a dependency, many folks are using libp2p independently or embedded in other projects such as X, Y, and Z (insert examples that don't use IPFS or Filecoin).
```

Comment:
```suggestion
Historically, companies that produced peer-to-peer applications like like Skype or BitTorrent created their own protocols to support them. Those protocols made a lot of assumptions about the environments they'd run in and the needs they'd meet, making them incredibly hard to upgrade or adapt to new environments. libp2p, however, has been extracted from its original implementation as the networking stack of IPFS, and can now serve a broad variety of use cases. While everyone building on IPFS is using libp2p as a dependency, many folks are using libp2p independently or embedded in other projects such as X, Y, and Z (insert examples that don't use IPFS).
```

Since Filecoin uses IPFS, maybe it's not necessary to mention it + IPFS is more of a dev tool than Filecoin I believe, so just using IPFS sounds better.
Just 2 cents.

Reply:
@zebateira I'm inclined to leave this because (a) not everyone knows Filecoin uses IPFS and (b) libp2p sounds like it's more popular if we include both. @yusefnapora What do you think? Could either of you help with these examples I didn't insert? 😂